Smrtconnect: Add new bidder

psmrt opened the pull request:
I have added adapter for smrtconnect

SerhiiNahornyi commented on src/test/java/org/prebid/server/bidder/smrtconnect/SmrtconnectBidderTest.java:
```diff
@@ -0,0 +1,115 @@
+    }
+
+    private static BidRequest givenBidRequest(
+            Function<BidRequest.BidRequestBuilder, BidRequest.BidRequestBuilder> bidRequestCustomizer,
```
Please use `UnaryOperator` instead of `Function`

psmrt replied:
@SerhiiNahornyi Thanks for review i have updated pull request.

SerhiiNahornyi commented on src/test/java/org/prebid/server/bidder/smrtconnect/SmrtconnectBidderTest.java:
```diff
@@ -0,0 +1,115 @@
+        assertThat(result.getErrors())
+                .containsExactly(BidderError.badServerResponse("Bad Server Response"));
+    }
+
```
Please add test for each `mtype`

SerhiiNahornyi commented on src/test/java/org/prebid/server/it/SmrtconnectTest.java:
```diff
@@ -0,0 +1,36 @@
+    public void openrtb2AuctionShouldRespondWithBidsFromTheSmrtconnect() throws IOException, JSONException {
+        // given
+        WIRE_MOCK_RULE.stubFor(post(urlPathEqualTo("/smrtconnect-exchange"))
+                .withRequestBody(equalToJson(jsonFrom("openrtb2/smrtconnect/test-smrtconnect-bid-request.json")))
```
You have added `supply_id` query param for your test url, so you can also add `.withQueryParam` check here

SerhiiNahornyi commented on src/main/java/org/prebid/server/proto/openrtb/ext/request/smrtconnect/ExtImpSmrtconnect.java:
```diff
@@ -0,0 +1,13 @@
+import lombok.AllArgsConstructor;
+import lombok.Value;
+
+@Value
+@AllArgsConstructor(staticName = "of")
```
@Value(staticConstructor = "of")

SerhiiNahornyi commented on src/main/java/org/prebid/server/proto/openrtb/ext/request/smrtconnect/ExtImpSmrtconnect.java:
```diff
@@ -0,0 +1,13 @@
+@AllArgsConstructor(staticName = "of")
+public class ExtImpSmrtconnect {
+
+    @JsonProperty("supply_id")
```
Snake case is default for PBS, `@JsonProperty("supply_id")` can be removed

SerhiiNahornyi commented on src/main/java/org/prebid/server/bidder/smrtconnect/SmrtconnectBidder.java:
```diff
@@ -0,0 +1,110 @@
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+    private static final String SUPPLY_ID_MACRO = "{{SupplyId}}";
+    private static final TypeReference<ExtPrebid<?, ExtImpSmrtconnect>> SMRTCONNECT_EXT_TYPE_REFERENCE =
```
Static class fields should be on top of class definition (before `endpointUrl` and `mapper`)
Check examples inside other bidders

psmrt commented:
> @psmrt - since you don't support TCF, you probably don't run in Europe.
> 
> Please see https://docs.prebid.org/prebid-server/developers/add-new-bidder-go.html#consider-your-geography and consider adding a note like the one in 33across.yaml to your yaml file:
> 
> ```
> # This bidder does not operate globally. Please consider setting "disabled: true" in European datacenters.
> geoscope:
>   - "!EEA"
> ```
Thanks for review.
i have updated config file and updated pr with all requested change.
